### DISCUSSION TITLE
Change grid col-span-x

### DISCUSSION THREAD

**Discussion by TomDeSmet** [tailwindlabs/tailwindcss#3491] [Help] [answered]
I'm kind of lost why the `col-span-x` classes result in `grid-column: span x / span x`. Wouldn't it be better to have them output `grid-column-end: span x`?

They just need to span x columns. I didn't find any explanation anywhere what use-case `span x` has on `grid-column-start`. If anybody can share a link with an explanation of span 4 used as a starting point, please share or ellaborate.

I noticed this because I had the following code in my website:
`col-start-2 md:col-span-4` and the `col-span-4` overruled the start position of my first class because of higher specificity (media query). The `grid-column-start: span 4` starts at column 1.

**Comment by adamwathan** [comment, marked as answer]
Hey! Reason for this is so you can combine a span utility with either a start _or_ an end utility.

```html
<div class="col-span-4 col-start-2">
<div class="col-span-3 col-end-7">
```
